### PR TITLE
simplify process_stock and break up create_models_for_stock_report

### PR DESCRIPTION
**Pull request by dannyroberts**
also try to use fully descriptive names for variables

### CODE REVIEW

File: corehq/apps/commtrack/processing.py [hunk 1]
```diff
@@ -98,30 +107,29 @@ def process_stock(xform, case_db=None):
...
 
     # validate product ids
-    is_empty = lambda product_id: product_id is None or product_id == ''
-    if any([is_empty(tx.product_id) for tx in transactions]):
+    if any(transaction_helper.product_id in ('', None)
+            for transaction_helper in transaction_helpers):
         raise MissingProductId(
             _('Product IDs must be set for all ledger updates!'))
-    # transactions grouped by case/product id
-    grouped_tx = map_reduce(lambda tx: [((tx.case_id, tx.product_id),)],
```
Comment (dannyroberts) [inline corehq/apps/commtrack/processing.py]: These few lines are my favorite

**Comment by dannyroberts** [comment]
@czue @TylerSheffels

File: corehq/apps/commtrack/processing.py [hunk 2]
```diff
@@ -98,38 +107,39 @@ def process_stock(xform, case_db=None):
...
-                            lambda v: sorted(v, key=lambda tx: tx.timestamp),
-                            data=transactions,
-                            include_docs=True)
 
-    case_ids = list(set(k[0] for k in grouped_tx))
```
Comment (czue) [inline corehq/apps/commtrack/processing.py]: wow

Comment (czue) [inline corehq/apps/commtrack/processing.py]: i have to imagine/hope that `grouped_tx` used to be used a bit more

File: corehq/apps/commtrack/processing.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/commtrack/processing.py]: :astonished:

**Pull request merged**